### PR TITLE
kube-proxy: code optimization for readability

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig network

#### What this PR does / why we need it:
Better readability.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/topology.go [hunk 1]
```diff
@@ -57,11 +57,7 @@ func CategorizeEndpoints(endpoints []Endpoint, svcInfo ServicePort, nodeLabels m
 		// effort attempt to avoid dropping connections.
 		if len(clusterEndpoints) == 0 && utilfeature.DefaultFeatureGate.Enabled(features.ProxyTerminatingEndpoints) {
 			clusterEndpoints = filterEndpoints(endpoints, func(ep Endpoint) bool {
-				if ep.IsServing() && ep.IsTerminating() {
-					return true
-				}
-
-				return false
+				return ep.IsServing() && ep.IsTerminating()
```
Comment (aojea) [inline pkg/proxy/topology.go:60]: I prefer the old format, it is easier to read

File: pkg/proxy/service.go [hunk 2]
```diff
@@ -296,14 +296,15 @@ func NewServiceChangeTracker(makeServiceInfo makeServicePortFunc, ipFamily v1.IP
 // Delete item
 //   - pass <service, nil> as the <previous, current> pair.
 func (sct *ServiceChangeTracker) Update(previous, current *v1.Service) bool {
+	// This is unexpected, we should return false directly.
+	if previous == nil && current == nil {
```
Comment (aojea) [inline pkg/proxy/service.go:300]: indeed, this is better

File: pkg/proxy/topology.go (hunk 1 above)
Comment (kerthcet) [reply pkg/proxy/topology.go:60]: fair, updated.

**Pull request merged**